### PR TITLE
[Mime] deprecate attach/embed methods in favor of Email::addPart()

### PR DESCRIPTION
**Pull request by fabpot**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | yes 
| Tickets       | n/a 
| License       | MIT
| Doc PR        | -

#47462 follow-up

### CODE REVIEW

File: src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php [hunk 1]
```diff
@@ -38,23 +40,17 @@ public function toName(): string
...
-        } else {
-            $this->message->embed($file->getCode(), $image, $contentType);
-        }
+        $body = $file->getPath() ? new BodyFile($file->getPath()) : $file->getCode();
+        $this->message->addPart((new DataPart($body, $image, $contentType))->asInline());
 
         return 'cid:'.$image;
     }
 
     public function attach(string $file, string $name = null, string $contentType = null): void
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php:49]: should be deprecated also I guess?

File: src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php [hunk 2]
```diff
@@ -38,23 +40,17 @@ public function toName(): string
     public function image(string $image, string $contentType = null): string
```
Comment (nicolas-grekas) [inline src/Symfony/Bridge/Twig/Mime/WrappedTemplatedEmail.php:40]: what about this one?

**Review by nicolas-grekas — COMMENTED** [review]
Changelog entries missing :)

But I'm wondering: is it worth it? I mean: this will have an impact on all current users of the component, and will make it harder for bundles to provide compat with many versions at once (and be deprecation-free). Also, it's a bit longer to type and might be a bit harder to discover (`embed()` vs `->asInline()`). What's the issue with the current way of doing (and naming ;)) things ?

**Comment by fabpot** [comment]
As mentioned in the related PR, I'm not sure this is worth it either. That's just a proposal.
The current names are slightly wrong; that's why it might make sense to deprecate them.
But again, not 100% convinced.

**Comment by stof** [comment]
To make it easier to support multiple versions, we might keep those names around for now, and deprecate them only in 7.1. As those methods are implemented as shortcuts on top of the new API, their maintenance cost is quite low.

However, if you expect to rename `attachPart` to `addPart` as part of the deprecations, I would add the new name already.

**Comment by nicolas-grekas** [comment]
Thanks for the insights. I'm "-0" personnaly: not worth the trouble pushed to the community IMHO.

**Comment by fabpot** [comment]
I would at least keep the removal of using these methods internally and I would advocate for adding the addPart() method to replace the attachPart one.

**Comment by fabpot** [comment]
I've removed all deprecations but the one for `attachPart()` which is semantically wrong.

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**